tests/regression/test_issue_1711.c

```diff
@@ -0,0 +1,149 @@
+#include <string.h>
+#include <CL/cl.h>
+
+typedef struct {
+    cl_long* ptr;
+    size_t maxsize;
+    struct {
+        size_t dim1;
+        size_t dim2;
+    } dims;
+    size_t len;
+} DeviceArray;
+
+#define CHECK_ERROR(err) \
+    if (err != CL_SUCCESS) { \
+        printf("OpenCL Error %d at %s:%d\n", err, __FILE__, __LINE__); \
+        exit(1); \
+    }
+    
+cl_platform_id get_platform(int index) {
+    cl_uint num_platforms;
+    cl_int err = clGetPlatformIDs(0, NULL, &num_platforms);
+    CHECK_ERROR(err);
+
+    if (index >= num_platforms) {
+        printf("Platform index %d is out of range (max %d) at %s:%d\n",
+               index, num_platforms - 1, __FILE__, __LINE__);
+        exit(1);
+    }
+
+    cl_platform_id* platforms = (cl_platform_id*)malloc(sizeof(cl_platform_id) * num_platforms);
+    err = clGetPlatformIDs(num_platforms, platforms, NULL);
+    CHECK_ERROR(err);
+
+    cl_platform_id platform = platforms[index];
+    free(platforms);
+    return platform;
+}
+
+int main(int argc, char** argv) {
+    // Use first platform by default or take from command line
+    int platform_index = 0;
+    if (argc > 1) {
+        platform_index = atoi(argv[1]);
+    }
+
+    cl_int err;
+    cl_platform_id platform = get_platform(platform_index);
+    cl_device_id device;
+
+    // Get platform name
+    char platform_name[128];
+    err = clGetPlatformInfo(platform, CL_PLATFORM_NAME, sizeof(platform_name), platform_name, NULL);
+    CHECK_ERROR(err);
+
+    // Get device
+    err = clGetDeviceIDs(platform, CL_DEVICE_TYPE_ALL, 1, &device, NULL);
+    CHECK_ERROR(err);
+
+    // Get device name
+    char device_name[128];
+    err = clGetDeviceInfo(device, CL_DEVICE_NAME, sizeof(device_name), device_name, NULL);
+    CHECK_ERROR(err);
+
+    // Rest of the code remains the same...
+    cl_context context = clCreateContext(NULL, 1, &device, NULL, NULL, &err);
+    CHECK_ERROR(err);
+
+    cl_queue_properties props[] = {0};
+    cl_command_queue queue = clCreateCommandQueueWithProperties(context, device, props, &err);
+    CHECK_ERROR(err);
+
+    // Create DeviceArray structure
+    DeviceArray A;
+    A.maxsize = 64;
+    A.dims.dim1 = 8;
+    A.dims.dim2 = 1;
+    A.len = 8;
+
+    // Allocate SVM memory for the data array
+    A.ptr = (cl_long*)clSVMAlloc(context, CL_MEM_READ_WRITE, sizeof(cl_long) * A.len, 0);
+    if (A.ptr == NULL) {
+        printf("SVM allocation failed at %s:%d\n", __FILE__, __LINE__);
+        return 1;
+    }
+
+    // Load SPIR-V binary
+    FILE* f = fopen(SRCDIR "/test_issue_1711.spv", "rb");
+    if (!f) {
+        printf("Failed to open test_issue_1711.spv at %s:%d\n", __FILE__, __LINE__);
+        return 1;
+    }
+    fseek(f, 0, SEEK_END);
+    size_t size = ftell(f);
+    fseek(f, 0, SEEK_SET);
+
+    unsigned char* binary = (unsigned char*)malloc(size);
+    fread(binary, 1, size, f);
+    fclose(f);
+
+    cl_program program = clCreateProgramWithIL(context, binary, size, &err);
```

Comment:
What happens with this test if SPIR-V support is not enabled in the build? Seems all CIs pass so perhaps all of them have it enabled or? Can we make the test conditional on SPIR-V support in the cmake file?